core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/BillingApiService.scala

```diff
@@ -21,34 +21,40 @@ trait BillingApiService extends HttpService with PerRequestCreator with UserInfo
 
   val userServiceConstructor: UserInfo => UserService
 
-  val billingRoutes = requireUserInfo() { userInfo =>
-    path("billing" / Segment / "members") { projectId =>
-      get {
-        requestContext => perRequest(requestContext,
-          UserService.props(userServiceConstructor, userInfo),
-          UserService.GetBillingProjectMembers(RawlsBillingProjectName(projectId)))
+  val billingRoutes = detach() {
```

Comment:
What is `detach()`, and why does it need to go on every route?

Reply:
detach is used to execute the route in a future, I added it to every route that uses requireUserInfo since that makes a call out to sam.

http://spray.io/documentation/1.2.4/spray-routing/execution-directives/detach/

Reply:
f***ing spray. Ok thanks

Reply:
Could it be built in to `requireUserInfo` since it will always be required by it?

Reply:
I tried doing it that way but couldn't find a way to do so.

Reply:
try changing the beginning of the definition to this: `def requireUserInfo(): Directive1[UserInfo] = detach() & {`

Reply:
@jmthibault79  I am getting a warning when I make the above change.

```Users/vkumra/Documents/firecloud/rawls/core/src/main/scala/org/broadinstitute/dsde/rawls/openam/StandardUserInfoDirectives.scala:25: Adaptation of argument list by inserting () has been deprecated: this is unlikely to be what you want.
[warn]         signature: ExecutionDirectives.detach(dm: spray.routing.directives.DetachMagnet): spray.routing.Directive0
[warn]   given arguments: <none>
[warn]  after adaptation: ExecutionDirectives.detach((): Unit)
[warn]   def requireUserInfo(): Directive1[UserInfo] = detach() & {
[warn]                                                       ^
[warn] there were 28 feature warnings; re-run with -feature for details
[warn] two warnings found
```